### DISCUSSION TITLE
[Icon request] Add ChucK logo please

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [ryanoasis/nerd-fonts#1479] [Ideas]
Hi :)

As the title says, I would like to suggest the addition of an icon for [Chuck](https://chuck.cs.princeton.edu).

Logo is here: 
![Chuck logo](https://chuck.cs.princeton.edu/doc/images/chuck-logo2023w.png)

If I had the skills I'd try to patch and contribute. I hope this can be implemented.

Please and thank you

EDIT1:

just in case, this is how I set it for the time being. Mostly posting for the extension and name:

```
[...]
    {
      "nvim-tree/nvim-web-devicons",
      opts = {
        override_by_extension = {
          ["ck"] = {
            icon = "󰧚",
            color = "#80ff00",
            name = "ChucK",
          },
        },
      },
    },
[...]
```

**Comment by Finii** [comment]
Thanks for suggesting.

Do you have, by chance, just because I'm curious, an explanation why the github repo is at Stanford but the web page at Princeton?

**Reply by andreacfromtheapp** [reply]
they are mirrors. not sure about the internals. both official and valid.

**Comment by Finii** [comment]
SVG is here, quickly done using Inkscape and the original png image:

![chuck](https://github.com/ryanoasis/nerd-fonts/assets/16012374/6b007590-efbf-415f-a03f-cc2bab436544)

Maybe the square-brackets top and bottom are a bit thin?

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/bd18ff60-378c-4489-9348-38dbbeb24fd8)

**Reply by andreacfromtheapp** [reply]
great job :)

Personally, I think that ChucK operator (=>) is the most important elements here. Perhaps the square brakets could be elongated vertically a little to enclose it? (like the terminal with [>] does?). Just an idea. Not sure modifying a logo is allowed.

**Reply by andreacfromtheapp** [reply]
Hi @Finii :)

is there anything I can do to move this forward? I'm not familiar with the procedure/timing/rules. Just checking :)

Please and thank you

**Reply by Finii** [reply]
I believe there is nothing you can do, at least what I see.

There is not really a procedure. In the past people just threw in Issues (like you did) and then they sat there, for ... years.
In open state. We had like 500 Issues open, and of course noone will ever scroll down far.

Then I decided that this is no basis for working efficiently. I wormed through all open Issues and closed in the process almost all. Either fixed the bugs, tried to follow up, closed obsolete stuff etc.
I thought (for new icons) it is best:
* Create an Issue here (as you did)
* That issue is linked to in this meta Issue
  * #1095 
* This issue is closed
* When a new release is started 1095 is looked up, fonts and icons selected to really include etc
* Icon PRs created
* Release

The problem with new icons is that a) Nerd Fonts is intended as collection of collections. Not a icon collection itself. So usually new icons are added through other icon collections (only); for example Font Logos might be the more fitting place for the ChucK icon. But that means of course even longer time from request to release in Nerd Fonts.
And the b) there are some very few icons added directly by Nerd Fonts. That should really exceptional icons - because the number of icons (the codepoints) are very limited and we will use them all up in a year or so if all requested icons were added. So there is an additional hurdle that icons must take, being deemed so important that they must be added to the very limited own-Nerd-Fonts-Icon-Set. And who decides which icons are 'important'? 🤷‍♀️ Pure arbitrary rule :-(

So, what does that mean?
It does not look too bad for ChucK, at least I invested already some time here. On release time (3.2 is "expected" (also arbitrarly) before summer) I or some other maintainer (if there is one) checks all icon suggestions and PRs and arbitrarily selects some, maybe ChucK.

The problem I have is that I have noone to discuss. There are people here, but they all have some own-goal. They want something for themselves. Noone or very seldom is there who likes to discuss stuff from a Nerd Fonts point of view. 

Sorry if that is not the answer you expected. But anyhow, lets see where this carries us... 👍

**Reply by andreacfromtheapp** [reply]
Don't be sorry, this is an amazing answer. Complete and informative. Thank you :)

Should I link to the meta-issue? Or it something only contributors do?

Fingers crossed!

**Reply by andreacfromtheapp** [reply]
ps: in the meantime I've opened a Vim feature request to add native filetype support as well: https://github.com/vim/vim/issues/13886

**Reply by andreacfromtheapp** [reply]
and.... didn't mean to be rude: thank you for all of the above! I appreciate you <3

**Reply by Finii** [reply]
> didn't mean to be rude

I did not feel any rude-ness; I would not have taken so much time to answer then ;-)

I exaggerated the number of Issues, here is the bug life graph [1], and as you see I need to work harder to get down again...

### 10 Years of Nerd Fonts

<img width="909" alt="Screenshot 2024-01-18 at 22 50 23" src="https://github.com/ryanoasis/nerd-fonts/assets/16012374/db639764-713f-4ef6-95e0-8a060cf449d2">

[1] https://9-volt.github.io/bug-life

**Reply by andreacfromtheapp** [reply]
Just a quick note on this, as a "historical reason". I appreciate your work and keep my fingers crossed for this to be included at some point :)

> The problem with new icons is that a) Nerd Fonts is intended as collection of collections. Not a icon collection itself. So usually new icons are added through other icon collections (only); for example Font Logos might be the more fitting place for the ChucK icon.

I'e opened a request here, because I'm using `nvim-web-devicons` which in turns it uses Nerd Fonts. Had I known about Font Logs perhaps I should have requested there. You live you learn.

**Reply by Finii** [reply]
See #1489 scheduled for 3.2.0

**Comment by Finii** [comment]
Looks unclean:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/82b3212d-cab5-47e2-8960-a3ef4f17b1d0)

**Reply by Finii** [reply]
Fixed by drawing again by hand.

**Comment by andreacfromtheapp** [comment]
> See #1489 scheduled for 3.2.0

thank you!

**Comment by Finii** [comment]
Locking this to make sure discussion happen in the Issue thread.
